python/pyspark/sql/cogroup.py

```diff
@@ -78,6 +83,19 @@ def apply(self, udf):
         |20000101|  2|2.0|  y|
         |20000102|  2|4.0|  y|
         +--------+---+---+---+
+        >>> @pandas_udf("time int, id int, v1 double, v2 string", PandasUDFType.COGROUPED_MAP)  # doctest: +SKIP
```

Comment:
Can we insert some documentation like
https://github.com/apache/spark/blob/81de9d3c296da9e632376e720189674401bd6639/python/pyspark/sql/functions.py#L3100-L3113

BTW, I think we should document this pandas udf type later at `pandas_udf`'s docstring. This one can be done together with [SPARK-29126](https://issues.apache.org/jira/browse/SPARK-29126)